packages/apptive_grid_form/test/apptive_grid_form_test.dart

```diff
@@ -332,6 +371,43 @@ void main() {
         () => client.loadForm(uri: Uri.parse('/api/a/form')),
       ).called(2);
     });
+
+    testWidgets('Reload flag', (tester) async {
+      final target = TestApp(
+        client: client,
+        child: ApptiveGridForm(
+          uri: Uri.parse('/api/a/form'),
+        ),
+      );
+      final action = ApptiveLink(uri: Uri.parse('uri'), method: 'method');
+      final formData = FormData(
+        id: 'formId',
+        name: 'Form Name',
+        title: 'Form Title',
+        components: [],
+        links: {ApptiveLinkType.submit: action},
+        fields: [],
+        reloadAfterSubmit: true,
+      );
+
+      when(
+        () => client.loadForm(uri: Uri.parse('/api/a/form')),
+      ).thenAnswer((realInvocation) async => formData);
+      when(() => client.submitFormWithProgress(action, formData)).thenAnswer(
+        (_) => Stream.value(
+          SubmitCompleteProgressEvent(http.Response('', 200)),
+        ),
+      );
+
+      await tester.pumpWidget(target);
+      await tester.pumpAndSettle();
+      await tester.tap(find.byType(ActionButton));
+      await tester.pumpAndSettle();
+
+      verify(
```

Comment:
This should also check that that the success state is not shown but an empty form is shown

Reply:
done